packages/http/src/validator/__tests__/index.spec.ts

```diff
@@ -51,150 +43,112 @@ describe('HttpValidator', () => {
-          .mockReturnValueOnce({ body: false, headers: false, hijack: false, query: true });
-
-        expect(
-          httpValidator.validateInput({
-            resource: Object.assign(
-              {
-                method: 'get',
-                path: '/',
-                id: '1',
-                request: {},
-                responses: [{ code: '200' }],
-              },
-              extendResource,
-            ),
-            input: Object.assign({ method: 'get', url: { path: '/', query: {} } }, extendInput),
-            config: { cors: false, mock: { dynamic: false }, validate: { request: { query: true } } },
-          }),
-        ).toEqual([mockError]);
+  describe('headers validation in enabled', () => {
+    const test = (extendResource?: Partial<IHttpOperation>, length: number = 1) => () => {
+      expect(
+        httpValidator.validateInput({
+          resource: Object.assign(
+            {
+              method: 'get',
+              path: '/',
+              id: '1',
+              request: {},
+              responses: [{ code: '200' }],
+            },
+            extendResource,
+          ),
+          input: { method: 'get', url: { path: '/' } },
+          config: { cors: false, mock: { dynamic: false }, validateRequest: true, validateResponse: true },
+        }),
+      ).toHaveLength(length);
+    };
+
+    describe('request is not set', () => {
+      it('validates headers', test(undefined, 2));
+    });
+  });
 
-        expect(resolveValidationConfigModule.resolveRequestValidationConfig).toHaveBeenCalled();
-        expect(httpBodyValidator.validate).not.toHaveBeenCalled();
-        expect(httpHeadersValidator.validate).not.toHaveBeenCalled();
-        expect(httpQueryValidator.validate).toHaveBeenCalledWith({}, [], undefined);
-      };
+  describe('query validation in enabled', () => {
+    const test = (
+      extendResource?: Partial<IHttpOperation>,
+      extendInput?: Partial<IHttpRequest>,
+      lenght: number = 2,
+    ) => () => {
+      expect(
+        httpValidator.validateInput({
+          resource: Object.assign(
+            {
+              method: 'get',
+              path: '/',
+              id: '1',
+              request: {},
+              responses: [{ code: '200' }],
+            },
+            extendResource,
+          ),
+          input: Object.assign({ method: 'get', url: { path: '/', query: {} } }, extendInput),
+          config: { cors: false, mock: { dynamic: false }, validateRequest: true, validateResponse: true },
+        }),
+      ).toHaveLength(lenght);
+
+      expect(httpBodyValidator.validate).not.toHaveBeenCalled();
+      expect(httpHeadersValidator.validate).toHaveBeenCalled();
+      expect(httpQueryValidator.validate).toHaveBeenCalledWith({}, [], undefined);
+    };
+
+    describe('request is not set', () => {
+      it('validates query', test(undefined, undefined, 2));
```

Comment:
hm.. maybe in addition to `toHaveLength(length)`, we could also check for the result itself? Not only its `length`?

Reply:
I can, but it's not going to be that useful, since the result is an array of the same object `[mockResult, mockResult]` — the validation functions here are mocked and always returning the same stuff.

Reply:
oooh, didn't see this comment :) never mind!